src/passes/MultiPassKawaseBlurPass.ts

```diff
@@ -0,0 +1,280 @@
+
+	get levels(): number {
+
+		return this.downsamplingTargets.length;
+
+	}
+
+	set levels(value: number) {
+
+		if(this.levels === value) {
+
+			return;
+
+		}
+
+		const renderTarget = this.renderTarget;
+		const output = this.output;
+
+		this.dispose();
+		this.disposables.clear();
+		output.buffers.clear();
+
+		this.downsamplingTargets = [];
+		this.upsamplingTargets = [];
+
+		for(let i = 0; i < value; ++i) {
+
+			const target = renderTarget.clone();
+			target.texture.name = "Downsampling.Target" + i;
+			this.downsamplingTargets.push(target);
+			this.disposables.add(target);
+			output.setBuffer(target.texture.name.toUpperCase(), target);
+
+		}
+
+		this.upsamplingTargets.push(renderTarget);
+		output.setBuffer(MultiPassKawaseBlurPass.BUFFER_KAWASE, renderTarget);
+
+		for(let i = 0; i < value; ++i) {
+
+			const target = renderTarget.clone();
+			target.texture.name = "Upsampling.Target" + i;
+			this.upsamplingTargets.push(target);
+			this.disposables.add(target);
+			output.setBuffer(target.texture.name.toUpperCase(), target);
+
+		}
+
+		this.onResolutionChange(this.resolution);
+
+	}
+
+	protected override onInputChange(): void {
+
+		if(this.input.defaultBuffer === null || this.input.defaultBuffer.value === null) {
+
+			return;
+
+		}
+
+		const { type, colorSpace } = this.input.defaultBuffer.value;
+
+		for(const target of this.downsamplingTargets.concat(this.upsamplingTargets)) {
+
+			target.texture.type = type;
+			target.texture.colorSpace = colorSpace;
+			target.dispose();
+
+		}
+
+		if(this.input.frameBufferPrecisionHigh) {
+
+			this.downsamplingMaterial.defines.FRAME_BUFFER_PRECISION_HIGH = true;
+			this.upsamplingMaterial.defines.FRAME_BUFFER_PRECISION_HIGH = true;
+
+		} else {
+
+			delete this.downsamplingMaterial.defines.FRAME_BUFFER_PRECISION_HIGH;
+			delete this.upsamplingMaterial.defines.FRAME_BUFFER_PRECISION_HIGH;
+
+		}
+
+		this.downsamplingMaterial.needsUpdate = true;
+		this.upsamplingMaterial.needsUpdate = true;
+
+	}
+
+	protected override onResolutionChange(resolution: Resolution): void {
+
+		let { width, height } = resolution;
+
+		for(let i = 0; i < this.levels; ++i) {
+
+			width = Math.round(width * 0.5);
+			height = Math.round(height * 0.5);
+
+			this.downsamplingTargets[i].setSize(width, height);
+
+		}
+
+		for(let i = 0; i < this.levels; ++i) {
+
+			width = Math.round(width * 2);
+			height = Math.round(height * 2);
+
+			this.upsamplingTargets[i].setSize(width, height);
+
+		}
+
+	}
+
+	render(): void {
+
+		if(this.renderer === null || !this.input.defaultBuffer || this.input.defaultBuffer.value === null) {
+
+			return;
+
+		}
+
+		const renderer = this.renderer;
+		const downsamplingMaterial = this.downsamplingMaterial;
+		const upsamplingMaterial = this.upsamplingMaterial;
+
+		let previousBuffer = this.input.defaultBuffer.value;
+
+		let { width, height } = this.resolution;
+
+
+		// TODO: unsure about what to assign as the fullscreen material when the pass requires multiple
```

Comment:
also not sure about the following: "unsure about what to assign as the fullscreen material when the pass requires multiple"

Reply:
"when the pass requires multiple..." materials? It's fine like this. The `fullscreenMaterial` can be set to any material defined in the class declaration. Choosing a fullscreen material depends on the render task(s) that should be performed right after. Most passes only use one, but switching is fine too.